docs/blog/2019-08-30-building-the-net-third-and-grove-website-in-gatsby/index.md

```diff
@@ -0,0 +1,65 @@
+
+## Pain points that we came across (and solved)
+
+We’re going to get technical here because the main pain points were around custom modifications to get exactly what we wanted.
+
+**Styling. **One of the more contentious issues in the JavaScript community is styling. CSS in JS, just that term, stirs up some mighty opinions. This won’t address the rift, we are firmly on the side of ‘css in js is dope’. Using Emotion in our Gatsby project has been amazing. It has allowed us to maintain a very clean project structure, without sacrificing any of the flexibility of using SASS. We could have easily used SASS in this project as Gatsby has support for it. Instead of having matching co-located style files, we write local styles directly in the component where they are scoped. If you're unfamiliar with scoping styles, it allows you to write styles with no fear of them leaking to an unintended target somewhere else in the DOM.
+
+**Build Times.** One of the pain points in Gatsby can be build times. We initially built against our existing instance of Drupal. However, the Drupal source plugin for Gatsby pulls in all the data in JSON API by default. So our build times were … not fast. We had a decision to make. Spend time limiting what is exposed in Drupal’s JSON API, write queries with filters and includes (which the source plugin allows for), or opt for an entirely new Drupal build. We took this third path. Which immediately allowed us to control our build times without getting into writing delicate queries and blocking certain content from the API. The new Drupal instance is build to be headless first, which is important. Next, we built content types that were specifically made to be ingested by Gatsby. Allowing for more dynamic content creation. This works by matching content types in Drupal to React components. This allows for some really nice content workflows, but those are in the works and not in the scope of this post.
+
+**Dynamic Content Creation.** This is an interesting topic in Gatsby. Their project structure has a ‘pages’ directory where all of your routes live. So if I want a page at www.mysite.com/posts, I just create a post.js file in that folder and the route exists and renders that component. In most React projects, routing is a bit more involved than this. So this skips quite a bit of boilerplate code. However, it creates an issue. React is declarative. And I like to follow that in my project structure as well. With this pattern, I have no place to see my site structure. It's a tradeoff. In that same vein, programmatic page creation happens in a bit of an obfuscated way. You generate these in the ‘gatsby-node’ file with some exposed APIs for static page generation Gatsby gives you. This isn't a big problem, but hides some of what is nice about React projects that have routes, these routes render components, and the components clearly render blog posts based on said route.
```

Comment:
```suggestion
**Dynamic Content Creation.** This is an interesting topic in Gatsby. Their project structure has a ‘pages’ directory where all of your static routes live. So if I want a page at `www.mysite.com/about`, I can create a `about.js` file in that folder and the route exists and renders that component. In most React projects, routing is a bit more involved than this. So this skips quite a bit of boilerplate code. However, it creates an issue. React is declarative. And I like to follow that in my project structure as well. With this pattern, I have no place to see my site structure. It's a tradeoff. In that same vein, programmatic page creation happens in a bit of an obfuscated way. You generate these routes using a template component and the ‘gatsby-node.js’ file with some exposed APIs for page generation Gatsby gives you. This isn't a big problem, but hides some of what is nice about React projects that have routes: these routes render components, and the components clearly render blog posts based on said route.
```
`Posts` usually follows the template and gatsby-node approach, so the setup with the pages directory being automatic could be a bit confusing as worded. I added a bit more clarifying text, let me know what you think.